[알린]Vending step4 - PR

jinios opened the pull request:
step4 싱글톤 구현했습니다~

jinios committed:
Implement singleton

jinios committed:
Update Readme - step4

godrm commented on VendingMachineApp/VendingMachineApp/Controller.swift:
```diff
@@ -59,10 +59,10 @@ struct Controller {
+//    func vendingMachineDescription() -> String {
+//        let vending = VendingMachine(stockItems: self.setVendingMachineStock(unit: 3))
+//        return vending.showStockDefault()
+//    }
```
주석으로 코드를 지울꺼면 그냥 지우세요. 우리에게는 git 이 있잖아요

jinios replied:
아! 컨트롤러를 안쓰다보니 발견하지못했습니다 지우겠습니다ㅜㅜ

godrm commented on VendingMachineApp/VendingMachineApp/AppDelegate.swift:
```diff
@@ -12,7 +12,7 @@ import UIKit
+    var vending = VendingMachine.sharedInstance
 
     // 기존에 저장된 값에서 불러와서 VendingMachine 객체 인스턴스를 생성하기
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
```
위에서 sharedInstance로 만들었는데
아래 DataStorage에서 load (엇... road 아니고 load 겠죠?)해서 self.vending 만 바꾸면
다른 곳에서 sharedInstance에 접근하면 안되지 않나요? 싱글톤 의미가 있나요?

jinios replied:
무슨 이런 오타가ㅠㅠㅠ... 네 수정하겠습니다!

godrm reviewed (commented):
싱글톤으로 만들고나면 참조하는 곳에 영향이 있는지 확인해야 합니다.
지금 변경한 부분에서는 그게 없네요. 
appdelegate.vending 을 참조하는 건지 VendingMachine.sharedInstance 를 참조하는 건지 확실하게 확인해보세요.

jinios committed:
Rename method

jinios committed:
Add class function - singleton object

jinios committed:
Delete unused code

jinios committed:
Modify- save data, singleton object
Add class function for singleton object - loadData(), sharedVendingMachine()
Delete vendingMachine variable from AppDelegate, match singleton object with a variable in viewController
Modify archiving singleton object

jinios committed:
Update README - step4

jinios commented:
피드백 사항 수정했습니다. 
- 싱글톤 객체에 객체 리턴, 로드된 데이터로 바꿔주는 함수 추가
- AppDelegate의 vending 변수를 넘기면서 저장하면 빈 자판기가 저장되는 문제 수정

godrm commented:
수고하셨습니다. 
다음 미션을 하면서 만든 싱글톤 객체에 의존성을 줄이도록 DIP를 적용해서 개선해보세요.
